core-chart/templates/argocd/operator.yaml

```diff
@@ -0,0 +1,163 @@
+
+      timestamped_echo "==> Copying kubeconfig \"${cp_name}.kubeconfig\" to Argo CD pod \"$ARGOCD_POD\"..."
+      kubectl -n $NS cp "$kubeconfig" ${ARGOCD_POD}:/tmp
+      rm "$kubeconfig"
+
+      timestamped_echo "==> Registering Control Plane \"$cp_name\" as an Argo CD cluster..."
+      kubectl -n $NS exec $ARGOCD_POD -- argocd cluster add ${cp_name} -y --kubeconfig "$kubeconfig"
+      kubectl -n $NS exec $ARGOCD_POD -- rm "$kubeconfig"
+
+    {{- end }}
+
+    timestamped_echo "Creating applications..."
+    {{- range $app := (index .Values "argo-cd" "applications") }}
+
+      timestamped_echo "==> Creating application \"{{ $app.name }}\":"
+      timestamped_echo "    - project=\"{{ $app.project | default "default" }}\""
+      timestamped_echo "    - repoURL=\"{{ $app.repoURL }}\""
+      timestamped_echo "    - targetRevision=\"{{ $app.targetRevision | default "HEAD" }}\""
+      timestamped_echo "    - path=\"{{ $app.path | default "/" }}\""
+      timestamped_echo "    - destinationWDS=\"{{ $app.destinationWDS }}\""
+      timestamped_echo "    - destinationNamespace=\"{{ $app.destinationNamespace | default "default" }}\""
+
+      kubectl exec $ARGOCD_POD -- argocd app create {{ $app.name }} \
+        --project {{ $app.project | default "default" }} \
+        --repo {{ $app.repoURL }} \
+        --revision {{ $app.targetRevision | default "HEAD" }} \
+        --path {{ $app.path | default "/" }} \
+        --dest-name {{ $app.destinationWDS }} \
+        --dest-namespace {{ $app.destinationNamespace | default "default" }} \
+        --sync-option CreateNamespace=true \
+        --sync-policy {{ $app.syncPolicy | default "manual" }} || true
+
+    {{- end }}
+
+    timestamped_echo "Done."
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-argocd-kubestellar-operator"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-weight": "5" # IMPORTANT: set this weight higher than KubeFlex post-install hook, otherwise postgresql will not install
+    "helm.sh/hook-delete-policy": before-hook-creation, hook-succeeded
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never # OnFailure
+      securityContext:
+        runAsNonRoot: true
+      serviceAccountName: "{{ .Release.Name }}-argocd-kubestellar-operator"
+      containers:
+      - name: add-wdses-clusters

```

Comment:
FYI, I don't think you need two plurals in this name. I think "add-wds-clusters" would be fine.